### ISSUE TITLE
CI tests error with Astropy 6.0

### ISSUE THREAD

**Issue by olebole** [aplpy/aplpy#492]
With Astropy 6.0, the CI tests do not start due to the following error:
```python
________________________ ERROR collecting test session _________________________
/usr/lib/python3/dist-packages/_pytest/config/__init__.py:641: in _importconftest
    mod = import_path(conftestpath, mode=importmode, root=rootpath)
/usr/lib/python3/dist-packages/_pytest/pathlib.py:567: in import_path
    importlib.import_module(module_name)
/usr/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1204: in _gcd_import
[…]
/usr/lib/python3/dist-packages/aplpy/__init__.py:9: in <module>
    from ._astropy_init import *  # noqa
/usr/lib/python3/dist-packages/aplpy/_astropy_init.py:21: in <module>
    from astropy.config.configuration import (
E   ImportError: cannot import name 'update_default_config' from 'astropy.config.configuration' (/usr/lib/python3/dist-packages/astropy/config/configuration.py)
```
Reason is that **update_default_config** was deprecated in Astropy 5.0 and [removed in Astropy 6.0](https://github.com/astropy/astropy/blob/main/CHANGES.rst#astropyconfig-1):
> * Removed deprecated ConfigurationMissingWarning class and update_default_config function; There are no replacements as they should no be used anymore. [#15466]

**Comment by pllim** [comment]
Looks like this repo has not been touched for a year. A lot of things would need upgrading besides this incompatibility.

I do not think this package uses astropy.config at all, so that whole setup could be removed.

**Comment by astrofrog** [comment]
I'll try and update the package infrastructure soon and will fix this at the same time

**Comment by Hellseher** [comment]
Hi, 

I experience the same after updating python-astropy to v6.0.0 in Guix toolchain.
Here is the input dependencies list:

- python-astropy@6.0.0
- python-matplotlib@3.8.2
- python-numpy@1.23.2
- python-pillow@9.2.0
- python-pyavm@0.9.6
- python-pyregion@2.2.0
- python-pytest-astropy@0.11.0
- python-pytest-mpl@0.16.1
- python-reproject@0.13.0
- python-scikit-image@0.22.0
- python-semantic-version@2.8.5
- python-shapely@2.0.1

Thanks,
Oleg

**Comment by Jing-Guo94** [comment]
> Hi,
> 
> I experience the same after updating python-astropy to v6.0.0 in Guix toolchain. Here is dependencies list:
> 
> * python-astropy@6.0.0
> * python-matplotlib@3.8.2
> * python-numpy@1.23.2
> * python-pillow@9.2.0
> * python-pyavm@0.9.6
> * python-pyregion@2.2.0
> * python-pytest-astropy@0.11.0
> * python-pytest-mpl@0.16.1
> * python-reproject@0.13.0
> * python-scikit-image@0.22.0
> * python-semantic-version@2.8.5
> * python-shapely@2.0.1
> 
> Thanks, Oleg

I updated all the dependencies to the same with you.  But unfortunately, it doesnot work in my case.

**Comment by Hellseher** [comment]
Hi @Jing-Guo94,

I might word it wrong - I have the same issue with tests.
`E   ImportError: cannot import name 'update_default_config' from 'astropy.config.configuration' `

```
________________________ ERROR collecting test session _________________________
/gnu/store/4r7k7ipiaqkdf4lmnxwmbz0wx2yzygzc-python-3.10.7/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:992: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:241: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:883: in exec_module
    ???
<frozen importlib._bootstrap>:241: in _call_with_frames_removed
    ???
aplpy/__init__.py:9: in <module>
    from ._astropy_init import *  # noqa
aplpy/_astropy_init.py:21: in <module>
    from astropy.config.configuration import (
E   ImportError: cannot import name 'update_default_config' from 'astropy.config.configuration' (/gnu/store/qhfqv8f3lhvw0n6l92yirhrnb9ll1qdp-python-astropy-6.0.0/lib/python3.10/site-packages/astropy/config/configuration.py)
=========================== short test summary info ============================
ERROR  - ImportError: cannot import name 'update_default_config' from 'astrop...
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
```

**Comment by Jing-Guo94** [comment]
> Hi @Jing-Guo94,
> 
> I might word it wrong - I have the same issue with tests. `E ImportError: cannot import name 'update_default_config' from 'astropy.config.configuration' `
> 
> ```
> ________________________ ERROR collecting test session _________________________
> /gnu/store/4r7k7ipiaqkdf4lmnxwmbz0wx2yzygzc-python-3.10.7/lib/python3.10/importlib/__init__.py:126: in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
> <frozen importlib._bootstrap>:1050: in _gcd_import
>     ???
> <frozen importlib._bootstrap>:1027: in _find_and_load
>     ???
> <frozen importlib._bootstrap>:992: in _find_and_load_unlocked
>     ???
> <frozen importlib._bootstrap>:241: in _call_with_frames_removed
>     ???
> <frozen importlib._bootstrap>:1050: in _gcd_import
>     ???
> <frozen importlib._bootstrap>:1027: in _find_and_load
>     ???
> <frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
>     ???
> <frozen importlib._bootstrap>:688: in _load_unlocked
>     ???
> <frozen importlib._bootstrap_external>:883: in exec_module
>     ???
> <frozen importlib._bootstrap>:241: in _call_with_frames_removed
>     ???
> aplpy/__init__.py:9: in <module>
>     from ._astropy_init import *  # noqa
> aplpy/_astropy_init.py:21: in <module>
>     from astropy.config.configuration import (
> E   ImportError: cannot import name 'update_default_config' from 'astropy.config.configuration' (/gnu/store/qhfqv8f3lhvw0n6l92yirhrnb9ll1qdp-python-astropy-6.0.0/lib/python3.10/site-packages/astropy/config/configuration.py)
> =========================== short test summary info ============================
> ERROR  - ImportError: cannot import name 'update_default_config' from 'astrop...
> !!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
> ```

Oh sorry, I misread it. I degrade the astropy version to 5.3.4, and the issue was solved. I hope this helps.

**Comment by Jing-Guo94** [comment]
> Hi @Jing-Guo94 ,
> 
> No worries.
> 
> I thought the topic of this issue was to highlight that apply has some incompatibility with the latest Astropy.
> 
> Thanks, Oleg

Oh yes!!! Finally, I realize this is not a topic ask for help. Any way, thanks!

**Comment by pllim** [comment]
Until this package can be patched to be compatible, you have to downgrade astropy for now, as pointed out in https://github.com/aplpy/aplpy/issues/492#issuecomment-1922601132 above.

**Comment by olebole** [comment]
Pragmatically, one can just remove the code block starting with https://github.com/aplpy/aplpy/blob/263435086747be37741619e0ec8cfa6c9e5664e1/aplpy/_astropy_init.py#L18

and still have all tests running and passing. However, to be honest I am not sure what the "cnfiguration update" actually does and whether it is required.

**Comment by jrthorstensen** [comment]
Thanks, Ole.

I also have a system running an older version of 
astropy on which aplpy still works for the moment, but if that
fails I'll certainly start fiddling with aplpy.

At the moment I use aplpy for a semi-automated procedure for
generating finding charts.  It's vastly less laborious than 
doing it by hand with ds9 or something, and makes nice charts,
too.

John

On Mon, 12 Feb 2024, Ole Streicher wrote:

> 
> Pragmatically, one can just remove the code block starting withhttps://github.com/aplpy/aplpy/blob/263435086747be37741619e0ec8cfa6c9e5664e
> 1/aplpy/_astropy_init.py#L18
> 
> and still have all tests running and passing. However, to be honest I am not
> sure what the "cnfiguration update" actually does and whether it is
> required.
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because youcommented.[ACB7JFKGKAZRM3BVQ7DVPZ3YTHQUPA5CNFSM6AAAAAA75K7R2WWGG33NNVSW45C7
> OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTTTRC5AG.gif] Message ID:
> ***@***.***>
> 
> 
>

--------------------------------------------------------
John R. Thorstensen
Department of Physics and Astronomy
Dartmouth College
6127 Wilder Laboratory
Hanover, NH 03755

Office: 239 Wilder Hall
Phone:  603 646 2869     Fax: 1446   Hinman Box 6127
--------------------------------------------------------

---1434368998-892075516-1707744603=:2727388--

**Comment by pllim** [comment]
tl;dr -- Yes, I think that config stuff can be removed. However, back in Nov 2023, @astrofrog also wanted to update a bunch of stuff along with removing it.

https://github.com/aplpy/aplpy/issues/492#issuecomment-1830115290

**Comment by dhomeier** [comment]
Seems some people are still having this problem with 5.3 (would probably be 5.3.4) on anaconda:
https://stackoverflow.com/questions/77973744/cannot-import-name-update-default-config-from-astropy-config-configuration

**Comment by pllim** [comment]
Looks like the SO user used full Anaconda install, not a fresh env. I won't waste too much time on that one. User should create a fresh env and try again. With the full install, it comes with a lot of baggage that can potentially silently downgrade stuff and so on.
